tests/models/test_trigger.py

```diff
@@ -142,10 +142,12 @@ def test_submit_failure(session, create_task_instance):
     assert updated_task_instance.next_method == "__fail__"
 
 
-def test_assign_unassigned(session, create_task_instance):
+def test_assign_unassigned(session, create_task_instance, monkeypatch):
     """
     Tests that unassigned triggers of all appropriate states are assigned.
     """
+    # disable QueueListener to prevent logging during tests
+    monkeypatch.setattr("airflow.jobs.triggerer_job_runner.DISABLE_LISTENER", True)
```

Comment:
Should not you restore it when the test completes? This creates side effect that might impact other tests later.

Reply:
Without this monkeypatch the console logging gets turned on for the tests that run after test_trigger when running all the Core tests together. The monkeypatch should undo the setattr once the fixture goes out of scope - https://docs.pytest.org/en/8.0.x/_modules/_pytest/monkeypatch.html

Reply:
> This creates side effect that might impact other tests later.

This might create only if we touch other parts which read this variable during test running, as we have before.

E.g. `attr_2` depend on `attr_1`. Test_1 mokeypatch `attr_1` and somehow involve `attr_2`, after execution `attr_1` return it previous value, but not for `attr_2`. Test 2 run test and use `attrt_2` the value unexpected and test failed.

Reply:
> This might create only if we touch other parts which read this variable during test running, as we have before.

That makes sense. It's just without setting the value here TriggererJobRunner ends up calling setup_queue_listener() which changes the logging and seemingly affects subsequent test modules.